src/app/zeko/da-layer/da_utils/da_utils.ml

```diff
@@ -0,0 +1,58 @@
+(** 
+  These functions call external C functions that call rust functions defined in rust library
+  These are long running functions with IO operations, so they are run in separate threads,
+  C code calls `caml_release_runtime_system` so that other threads can run in parallel.    
+*)
+open Async
+
+open Core
+
+external caml_post_batch :
+  string -> string -> string -> string -> string list -> (string, string) result
+  = "caml_post_batch"
+
+let post_batch ~da_websocket ~da_contract_address ~da_private_key ~batch_data
+    ~sig_data : (string, Error.t) result Deferred.t =
+  In_thread.run (fun () ->
+      caml_post_batch da_websocket da_contract_address da_private_key batch_data
+        sig_data
+      |> Result.map_error ~f:(fun e -> Error.of_string e) )
+
+external caml_get_batch_data :
+  string -> string -> string -> (string, string) result = "caml_get_batch_data"
+
+let get_batch_data ~da_websocket ~da_contract_address ~location :
+    (string, Error.t) result Deferred.t =
+  In_thread.run (fun () ->
+      caml_get_batch_data da_websocket da_contract_address location
+      |> Result.map_error ~f:(fun e -> Error.of_string e) )
+
+external caml_init_genesis_state :
+  string -> string -> string -> string -> (unit, string) result
+  = "caml_init_genesis_state"
+
+let init_genesis_state ~da_websocket ~da_contract_address ~da_private_key ~data
+    : (unit, Error.t) result Deferred.t =
+  In_thread.run (fun () ->
+      caml_init_genesis_state da_websocket da_contract_address da_private_key
+        data
+      |> Result.map_error ~f:(fun e -> Error.of_string e) )
+
+external caml_get_genesis_state : string -> string -> (string, string) result
+  = "caml_get_genesis_state"
+
+let get_genesis_state ~da_websocket ~da_contract_address :
```

Comment:
What is this for? What is genesis state?

Reply:
The state the rollup is starting for. It's to allow us not to throw away the current state.

Reply:
You don’t want this, rather you want a way of writing how the ledger changed as noted in #12

Reply:
yeah that's not ideal, we can however change it in future since the `Batch.t` is versioned